envoy/upstream/upstream.h

```diff
@@ -1192,6 +1198,16 @@ class ClusterInfo : public Http::FilterChainFactory {
    */
   virtual Http::HeaderValidatorPtr makeHeaderValidator(Http::Protocol protocol) const PURE;
 
+  /**
+   * Setting the cluster that owning this info, only for dynamic forward proxy cluster.
+   */
+  virtual void cluster(ClusterSharedPtr) const PURE;
+
+  /**
+   * @return the cluster that owning this info, only for dynamic forward proxy cluster.
+   */
+  virtual ClusterSharedPtr cluster() const PURE;
```

Comment:
I still don't understand why you need any interface changes. I don't think any should be required. You are going to be making a new cluster manually given the configuration which you should have access to from the DFP configuration. Can you please remove all of these changes or explain why they are needed?

Reply:
It's used to get the `DFP cluster` in `DFP http filer`:
https://github.com/envoyproxy/envoy/pull/26420/files#diff-960724c79345d5bd91d33c3108b549b8c50ca588c7bc57d65f359df841edacfcR272-R277

in the `DFP http filter`, we only have the DFP cluster name, and its `const cluster_info`.
this `cluster` api is used to get the `DFP cluster` from its `const cluster_info`.

why we need the `DFP cluster` in `DFP http filter`?
1. we need the new sub cluster configuration, inherit from the `DFP cluster`. since the `DFP http filter` doesn't have the cluster configurations.
    see: https://github.com/envoyproxy/envoy/pull/26420/files#diff-960724c79345d5bd91d33c3108b549b8c50ca588c7bc57d65f359df841edacfcR65-R66

2. we store the sub cluster name into `cluster_map_` in DFP cluster, so that we can limit the max number of sub clusters, and purge them when they are expired.
    see: https://github.com/envoyproxy/envoy/pull/26420/files#diff-8f0201f1ad64a4d5abf5d3cfe3388d19feea791341f62ebc32064622267da65aR96-R111

Or maybe, is there another proper way to get the `DFP cluster` in `DFP http filter`?
I know the `cluster` API is bit hacky here, but, this is the most proper way I figure out for now.
suggestions are welcome, thank you very much.

Reply:
Oh, I see, thanks.